pkg/controller/promotionreconciler/reconciler.go

```diff
@@ -195,24 +195,26 @@ func (r *reconciler) promotionConfig(ist *imagev1.ImageStreamTag) (*cioperatorap
 	}
 }
 
-func commitForIST(ist *imagev1.ImageStreamTag) (string, error) {
+// CommitForIST returns the commit for an isTag
+func CommitForIST(ist *imagev1.ImageStreamTag) (string, error) {
 	metadata := &docker10.DockerImage{}
 	if err := json.Unmarshal(ist.Image.DockerImageMetadata.Raw, metadata); err != nil {
 		return "", fmt.Errorf("failed to unmarshal imagestream.image.dockerImageMetadata: %w", err)
 	}
 
 	commit := metadata.Config.Labels["io.openshift.build.commit.id"]
 	if commit == "" {
-		return "", controllerutil.TerminalError(errors.New("ImageStreamTag has no `io.openshift.build.commit.id` label, can't find out source commit"))
```

Comment:
We do not need `controllerutil.TerminalError` because its caller has it already.